src/components/Modals/Relationship/StepOne.jsx

```diff
@@ -1,24 +1,107 @@
+/* eslint-disable jsx-a11y/click-events-have-key-events */
+import { useState, useContext } from "react";
+import { useQuery } from "@tanstack/react-query";
+
+import fetchData from "../../../utils/axios";
+
+import UserContext from "../../../context/UserContext";
+import CurrentDBIdContext from "../../../context/CurrentDBIdContext";
 import Content from "../SharedItems/Content";
 import Title from "../SharedItems/Title";
 import Button from "../../shared/Button";
 import Message from "../SharedItems/Message";
+import Loading from "../../shared/Loading";
+
+function StepOne({
+  setRelationshipStep,
+  databaseName,
+  relationData,
+  setRelationData,
+}) {
+  const [targetDatabases, setTargetDatabases] = useState([]);
+  const [isSelected, setIsSelected] = useState("");
+
+  const { userId } = useContext(UserContext);
+  const currentDBId = useContext(CurrentDBIdContext);
+
+  async function getDatabaseList() {
+    const response = await fetchData("GET", `users/${userId}/databases`);
+
+    return response.data.databases;
+  }
+
+  const { isLoading } = useQuery(["dbs"], getDatabaseList, {
+    enabled: !!userId,
+    onSuccess: result => {
+      const filteredDbs = result.filter(
+        database => database._id !== currentDBId,
+      );
+
+      setTargetDatabases(filteredDbs);
+    },
+    onFailure: () => {
+      console.log("sending user to errorpage");
+    },
+    refetchOnWindowFocus: false,
+  });
+
+  if (isLoading) {
+    return <Loading />;
+  }
+
+  function handleNextClick() {
+    if (!relationData.foreignDbId) {
+      alert("Please choose database to make relationship!");
+
+      return;
+    }
+
+    setRelationshipStep("stepTwo");
+  }
+
+  function handleDatabaseClick(id) {
+    setIsSelected(id);
+
+    setRelationData({
+      ...relationData,
+      foreignDbId: id,
+    });
+  }
 
-function StepOne({ setRelationshipStep }) {
   return (
     <>
       <Title>Step 1</Title>
       <Message>
         <p>Please choose a database that you would like to link with DBNAME</p>
       </Message>
       <Content>
-        <div className="flex justify-center items-center h-60 w-60">
-          database card here
+        <div className="flex flex-col justify-around items-center h-auto w-60">
+          <div className="flex border-2 rounded-lg justify-center items-center w-full p-2 bg-blue bg-opacity-50">
+            <p>{databaseName}</p>
+          </div>
+          <div className="border border-blue border-dashed h-16"></div>
+          <div className="flex flex-col border-2 rounded-lg items-center w-full max-h-[190px] overflow-y-scroll">
+            <ul className="w-full h-auto text-center">
+              {targetDatabases.map((database, index) => (
+                <li
+                  className={`w-full py-1 border-b-2 border-grey
+                  ${isSelected === database._id ? "bg-yellow" : ""}
+                  ${index === targetDatabases.length - 1 ? "border-b-0" : ""}`}
```

Comment:
```suggestion
                  ${isSelected === database._id && "bg-yellow"}
                  ${index === targetDatabases.length - 1 && "border-b-0"}`}
```

제가 이 비슷한 로직을 함수로 따로 뺐던 것 같은데.. 이거 간편하고 좋네요!
적용가능할지 봐야겠네요!! 👍

Reply:
안그래도 만들어 놓으셨던 함수 재사용 하려고 했었는데 개인적으론 함수로 빼는것보다 state로 관리해도 괜찮을듯 하여 일단은 onClick 함수 내부에 넣어 주었습니다 ㅎㅎ